x-pack/plugins/security_solution/public/detection_engine/rule_creation/components/required_fields/translations.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+export const REQUIRED_FIELDS_LABEL = i18n.translate(
+  'xpack.securitySolution.detectionEngine.ruleDescription.requiredFields.requiredFieldsLabel',
+  {
+    defaultMessage: 'Required fields',
+  }
+);
+
+export const FIELD_NAME = i18n.translate(
+  'xpack.securitySolution.detectionEngine.ruleDescription.requiredFields.fieldNameLabel',
+  {
+    defaultMessage: 'Field name',
+  }
+);
+
+export const FIELD_TYPE = i18n.translate(
+  'xpack.securitySolution.detectionEngine.ruleDescription.requiredFields.fieldTypeLabel',
+  {
+    defaultMessage: 'Field type',
+  }
+);
+
+export const REQUIRED_FIELDS_GENERAL_WARNING_TITLE = i18n.translate(
+  'xpack.securitySolution.detectionEngine.ruleDescription.requiredFields.generalWarningTitle',
+  {
+    defaultMessage: 'Some fields are not found within specified index patterns.',
+  }
+);
+
+export const REQUIRED_FIELDS_GENERAL_WARNING_DESCRIPTION = i18n.translate(
+  'xpack.securitySolution.detectionEngine.ruleDescription.requiredFields.generalWarningDescription',
+  {
+    defaultMessage: `This doesn't break rule execution, but it might indicate that required fields were set incorrectly. Please check that indices specified in index patterns exist and have expected fields and types in mappings.`,
```

Comment:
Want to make it a little clearer what "specified index patterns" means — specified where?

I don't know if this component allows bold type, but I suggested it to help identify the UI element where the index patterns are specified. Maybe use quotes if bold isn't available.
```suggestion
    defaultMessage: 'Some fields aren't found within the rule's specified index patterns.',
  }
);

export const REQUIRED_FIELDS_GENERAL_WARNING_DESCRIPTION = i18n.translate(
  'xpack.securitySolution.detectionEngine.ruleDescription.requiredFields.generalWarningDescription',
  {
    defaultMessage: `This doesn't break rule execution, but it might indicate that required fields were set incorrectly. Please check that indices specified in the rule's **Source** index patterns exist and have expected fields and types in mappings.`,
```